xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -393,18 +398,20 @@ def map_returns(op: Operation) -> None:
         apply = op.parent_op()
         assert isinstance(apply, ApplyOp)
 
-        res = list(apply.res)[0]
-
-        if (len(res.uses) > 1) or (not isinstance(
-            (store := list(res.uses)[0].operation), StoreOp)):
-            warn("Only single store result atm")
-            return
+        for res in list(apply.res):
+            if (len(res.uses) > 1) or (not isinstance(
+                (store := list(res.uses)[0].operation), StoreOp)):
+                warn("Only single store for a single return op result atm")
+                return
 
-        cast = store.field.owner
+            cast = store.field.owner
 
-        assert isinstance(cast, CastOp)
+            assert isinstance(cast, CastOp)
 
-        return_targets[op] = cast
+            if return_targets.get(op):
+                return_targets[op].append(cast)
+            else:
+                return_targets[op] = [cast]
```

Comment:
I think initializing `return_targets[op]` to `[]` before looping on results would be more readable. What do you think?